### PR TITLE
Apply projectHealth recommendations to various JVM testing deps

### PR DESCRIPTION
**Pull request by tresat**
Did these as a single PR because they:

- Required new verification metadata for bsh (used by testng).
- Required new explicit lib for junit platform engine.

This addresses the following projects:

  - plugins-jvm-test-suite
  - plugins-test-report-aggregation
  - testing-junit-platform
  - testing-jvm-infrastructure
  - testing-jvm
  - test-suites-base
  - testing-base

### CODE REVIEW

**Comment by tresat** [comment]
@bot-gradle test this

**Comment by tresat** [comment]
These tests are are CI issues that can be ignored - this is ready for review

**Comment by tresat** [comment]
@bot-gradle test this

File: gradle/verification-metadata.xml [hunk 1]
```diff
@@ -1227,6 +1227,11 @@
             <pgp value="BE685132AFD2740D9095F9040CC0B712FEE75827"/>
          </artifact>
       </component>
+      <component group="org.beanshell" name="bsh" version="2.0b4">
+         <artifact name="bsh-2.0b4.jar">
+            <sha256 value="91395c07885839a8c6986d5b7c577cd9bacf01bf129c89141f35e8ea858427b6" origin="Generated by Gradle" reason="Artifact is not signed"/>
```
Comment (cobexer) [inline gradle/verification-metadata.xml:1232]: This must be changed, see top of the file

**Review by cobexer — CHANGES_REQUESTED** [review]

File: gradle/verification-metadata.xml (hunk 1 above)
Comment (tresat) [reply gradle/verification-metadata.xml:1232]: @cobexer - Thanks for pointing this out.  I wasn't expecting to have to change anything here in this PR, so I was hoping someone would comment on this.

When I read the comment, it says run `./gradlew -i -write-verification-metadata sha256,pgp -export-keys resolveAllDependencies`.  If I delete my addition to this file, running that command recreates it as is.

[Looking at this artifact in Maven Central](https://repo1.maven.org/maven2/org/beanshell/bsh/2.0b4/) I don't see an `.asc` file, so I'm not sure what more to do here.

Any ideas?

**Comment by tresat** [comment]
@bot-gradle test this

File: platforms/jvm/testing-junit-platform/build.gradle.kts [hunk 2]
```diff
@@ -11,13 +11,14 @@ We make sure to include this subproject as a runtime dependency in :distribution
...
-    implementation(project(":logging"))
+    api(project(":base-annotations"))
+    api(project(":base-services"))
+    api(project(":messaging"))
+    api(project(":testing-base"))
+    api(project(":testing-jvm-infrastructure"))
 
-    implementation(libs.junitPlatform)
+    api(libs.junitPlatform)
+    api(libs.junitPlatformEngine)
```
Comment (big-guy) [inline platforms/jvm/testing-junit-platform/build.gradle.kts:21]: 🤔 Was this a missing direct dependency before?

**Review by big-guy — APPROVED** [review]

File: gradle/verification-metadata.xml (hunk 1 above)
Comment (tresat) [reply gradle/verification-metadata.xml]: Added notes about capabilities conflict after Louis shed some light on what was happening here.

File: platforms/jvm/testing-junit-platform/build.gradle.kts (hunk 2 above)
Comment (tresat) [reply platforms/jvm/testing-junit-platform/build.gradle.kts:21]: Yes, the plugin reported this as a transitive that should be promoted to direct.

**Comment by tresat** [comment]
@bot-gradle test and merge

**Pull request merged**